jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/sessionconfig/CodeScanSessionConfig.kt

```diff
@@ -64,7 +65,14 @@ sealed class CodeScanSessionConfig(
 
         LOG.debug { "Creating payload. File selected as root for the context truncation: ${selectedFile.path}" }
 
-        val (includedSourceFiles, payloadSize, totalLines, _) = includeDependencies()
+        val (includedSourceFiles, payloadSize, totalLines, _) = when (selectedFile.path.startsWith(projectRoot.path)) {
+            true -> includeDependencies()
+            false -> {
+                // Set project root as the parent of the selected file.
+                projectRoot = selectedFile.parent
+                includeFileOutsideProjectRoot()
```

Comment:
did we align this behavior that we will scan only the current folder in this case?

Reply:
It's only one file. Not even the entire folder. This is consistent with the current [VSC behavior](https://github.com/aws/aws-toolkit-vscode/blob/bdb0a832be5b48d1aae519bfc6602ef1a3229c13/src/codewhisperer/util/dependencyGraph/pythonDependencyGraph.ts#L180).